core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileBatchSource.java

```diff
@@ -102,41 +78,6 @@ public Schema getSchema(FileSourceConfig config, EndpointPluginContext pluginCon
       properties.put(PathTrackingInputFormat.COPY_HEADER, "true");
     }
 
-    // TODO:(CDAP-14424) Remove time table logic
-    // everything from this point on should be removed in a future release.
-    // the time table stuff is super specific, requiring input paths to be in a very specific format
-    // and it assumes the pipeline is scheduled to run in a specific way
-
-    //SimpleDateFormat needs to be local because it is not threadsafe
-    SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd-HH");
-
-    //calculate date one hour ago, rounded down to the nearest hour
-    Date prevHour = new Date(context.getLogicalStartTime() - TimeUnit.HOURS.toMillis(1));
-    Calendar cal = Calendar.getInstance();
-    cal.setTime(prevHour);
-    cal.set(Calendar.MINUTE, 0);
-    cal.set(Calendar.SECOND, 0);
-    cal.set(Calendar.MILLISECOND, 0);
-    prevHour = cal.getTime();
-
-    if (config.getTimeTable() != null) {
-      KeyValueTable table = context.getDataset(config.getTimeTable());
-      String datesToRead = Bytes.toString(table.read(LAST_TIME_READ));
-      if (datesToRead == null) {
-        List<Date> firstRun = new ArrayList<>(1);
-        firstRun.add(new Date(0));
-        datesToRead = GSON.toJson(firstRun, ARRAYLIST_DATE_TYPE);
-      }
-      List<Date> attempted = new ArrayList<>();
-      attempted.add(prevHour);
-      String updatedDatesToRead = GSON.toJson(attempted, ARRAYLIST_DATE_TYPE);
-      if (!updatedDatesToRead.equals(datesToRead)) {
-        table.write(LAST_TIME_READ, updatedDatesToRead);
-      }
-      properties.put(LAST_TIME_READ, datesToRead);
-    }
-
-    properties.put(CUTOFF_READ_TIME, dateFormat.format(prevHour));
     Pattern pattern = config.getFilePattern();
     properties.put(INPUT_REGEX_CONFIG, pattern == null ? ".*" : pattern.toString());
```

Comment:
I think we should remove this logic as well, I believe it hooks into the time table stuff